src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -246,6 +243,38 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
             return project;
         }
 
+        internal static string NormalizeProjectDirectoryPath(string path)
+        {
+            if (File.Exists(path) &&
```

Comment:
Agreed the readability could be improved, but I don't feel this is the right time to be making that refactoring. If you feel strongly, please log an issue for it.